bool on
bool up
bool down
bool left
bool right